stdsimd/arch/detect/arm.rs

```diff
@@ -35,9 +36,13 @@ pub enum Feature {
 pub fn detect_features() -> cache::Initializer {
     let mut value = cache::Initializer::default();
     fill_features(&mut value);
-    return value
+    value
 }
 
+#[cfg(not(target_os = "linux"))]
```

Comment:
I'm personally not a huge fan of `#[cfg]` as it can make already complicated code even more complicated, so could this continue to compile on all platforms and just statically return an error on the "wrong" platforms?

Reply:
IIRC the problem here was that the auxvec/cpuinfo modules are only available on some oses (currently only linux), but these functions should be available on all OSes and just return nothing on linux. 

So IMO the entanglement here is that we have target_os modules (e.g. linux) and arch modules (e.g. aarch64) and a mixture of archs and target_oses on both. I'll try to think of something better here.